### PR TITLE
Restore the old constructor to avoid incompatible issue

### PR DESCRIPTION
**Pull request by brycezhongqing**
As title

### CODE REVIEW

**Review by zackthehuman — APPROVED** [review]
Thanks for adding this back!

**Comment by zackthehuman** [comment]
This should fix the regression from https://github.com/linkedin/rest.li/pull/1027 with the constructors.

File: d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperConnectionManager.java [hunk 1]
```diff
@@ -184,6 +184,75 @@ protected None convertResponse(None none) throws Exception
     }
   }
 
+  /**
+   * deprecated, use {@link  ConnectionManager#markDownAllServers(Callback)} instead.
+   */
+  @Deprecated
+  public void markDownAllServers(final Callback<None> callback)
```
Comment (bohhyang) [inline d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperConnectionManager.java:191]: also needs "@Override", right? same for markUpAllServers

Comment (brycezhongqing) [reply d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperConnectionManager.java:191]: Actually. It no needs to add override, the java will get this automatically. But it's good to point it out

Comment (bohhyang) [reply d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperConnectionManager.java:192]: Yeah, just better for code readers.

**Review by bohhyang — APPROVED** [review]

**Pull request merged**